src/server/scripts/EasternKingdoms/zone_tirisfal_glades.cpp

```diff
@@ -38,134 +36,161 @@ EndContentData */
 
 enum Calvin
 {
-    SAY_COMPLETE        = 0,
-    SPELL_DRINK         = 2639,                             // possibly not correct spell (but iconId is correct)
-    QUEST_590           = 590
+    SAY_COMPLETE = 0,
+    SPELL_FOOD = 7737,
+    QUEST_590 = 590
+};
+
+enum event
+{
+    EVENT_EMOTE = 1,
+    EVENT_TALK = 2,
+    EVENT_DRINK_AND_QUEST_COMQUEST = 3,
+    EVENT_END = 4,
 };
 
 class npc_calvin_montague : public CreatureScript
 {
 public:
-    npc_calvin_montague() : CreatureScript("npc_calvin_montague") { }
+    npc_calvin_montague() : CreatureScript("npc_calvin_montague") {}
 
     bool OnQuestAccept(Player* player, Creature* creature, Quest const* quest) override
     {
         if (quest->GetQuestId() == QUEST_590)
         {
-            creature->SetFaction(FACTION_ENEMY);
-            creature->SetImmuneToPC(false);
-            CAST_AI(npc_calvin_montague::npc_calvin_montagueAI, creature->AI())->AttackStart(player);
+            creature->SetFaction(FACTION_MONSTER);
+            creature->RemoveNpcFlag(UNIT_NPC_FLAG_QUESTGIVER);
+            creature->RemoveUnitFlag(UNIT_FLAG_IMMUNE_TO_PC);//This is reset when out of combat
+            creature->SetUnitFlag(UNIT_FLAG_PET_IN_COMBAT);
+            creature->SetReactState(REACT_AGGRESSIVE);
+            creature->AI()->AttackStart(player);
         }
         return true;
     }
 
-    CreatureAI* GetAI(Creature* creature) const override
-    {
-        return new npc_calvin_montagueAI(creature);
-    }
-
     struct npc_calvin_montagueAI : public ScriptedAI
     {
-        npc_calvin_montagueAI(Creature* creature) : ScriptedAI(creature) { }
-
-        uint32 m_uiPhase;
-        uint32 m_uiPhaseTimer;
-        ObjectGuid m_uiPlayerGUID;
+        npc_calvin_montagueAI(Creature* creature) : ScriptedAI(creature) {}
 
         void Reset() override
         {
-            m_uiPhase = 0;
-            m_uiPhaseTimer = 5000;
-            m_uiPlayerGUID.Clear();
-
-            me->RestoreFaction();
-
-            if (!me->IsImmuneToPC())
-                me->SetImmuneToPC(true);
+            if (m_uiPhase != 1)
```

Comment:
m_uiPhase can be either 0 or 1, so use 
if (m_uiPhase) - This means true if m_uiPhase exist, aka any positive number
if (!m_uiPhase) - This means true if m_uiPhase does not exist, aka 0